process/input-parser.go

```diff
@@ -47,21 +48,33 @@ func parseInput(file string) ([]inputData, error) {
 		name = strings.TrimSpace(name)
 		zoneNames = append(zoneNames, name)
 	}
-	var dataArray []inputData
-	for {
-		var data inputData
-		var done bool
-		data, done, err = readOneRow(zoneNames, reader)
-		if done {
-			break
-		}
-		if err != nil {
-			klog.Infof("can't parse input data: %v, skip to next row\n", data.name)
-			continue
+	inputQueue := make(chan inputData)
+
+	go func() {
+		defer close(inputQueue)
+		defer func() {
+			cerr := inputFile.Close()
+			if cerr != nil {
+				klog.Errorf("close input file %s with an error %v", file, cerr)
+			}
+		}()
+
+		for {
+			var data inputData
+			var done bool
+			data, done, err = readOneRow(zoneNames, reader)
+			if done {
+				break
+			}
```

Comment:
Would something like this work?
```suggestion
		for data, done, err := readOneRow(zoneNames, reader); !done {
```

Reply:
changed to an embedded way: 
`for data, done, rerr := readOneRow(zoneNames, reader); !done; data, done, rerr = readOneRow(zoneNames, reader) {`
Need to call it again in the 3rd section and use another name other than err to avoid shadow previous err variable.